pkg/vulnsrc/oracle-oval/oracle-oval.go

```diff
@@ -183,16 +215,86 @@ func walkOracle(cri Criteria, osVer string, pkgs []AffectedPackage) []AffectedPa
 	return pkgs
 }
 
-func referencesFromContains(sources []string, matches []string) []string {
-	references := []string{}
-	for _, s := range sources {
-		for _, m := range matches {
-			if strings.Contains(s, m) {
-				references = append(references, s)
+func mergeVulnerabilityDetails(detail types.VulnerabilityDetail, oval OracleOVAL, vulnIDs []string) types.VulnerabilityDetail {
+	// Collect vulnerability details - references and severity
+	// A CVE can be present in multiple ELSAs.  Collect all the applicable references as we process them, later when done we'll insert
+	// the references.
+	convertedSeverity := severityFromThreat(oval.Severity)
+
+	// If multiple ELSAs for the same CVE have differing severities, use the highest one
+	if convertedSeverity > detail.Severity {
+		detail.Severity = convertedSeverity
+	}
+
+	for _, ref := range oval.References {
+		if referencesFromContains(ref.URI, vulnIDs) && !ustrings.InSlice(ref.URI, detail.References) {
+			detail.References = append(detail.References, ref.URI)
+		}
+	}
+
+	return detail
+}
+
+func mergeEntries(advisory Advisory, pkg AffectedPackage, elsaID string) Advisory {
+	affectedFlavor := GetPackageFlavor(pkg.Package.FixedVersion)
+
+	found := false
+	for i, entry := range advisory.Entries {
```

Comment:
This loop could be eliminated if we had a nested map here.  `bucket -> flavor -> Entry`.  I explored this but it resulted in random ordering of entries, which caused [tests](https://github.com/aquasecurity/trivy-db/pull/221/files#diff-c3694ef355672019be1ce23f5f5e6d131bfbd8c1c7e6cc93e7b1a9b8a0e6d90bR441) to always fail.  I didn't see another way around it other than sorting prior to saving the advisory details.

Reply:
I think your current version is good.